packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts

```diff
@@ -1,19 +1,150 @@
+/* eslint-disable no-console */
+import * as childproc from 'child_process';
+import * as fs from 'fs';
 import * as path from 'path';
 import * as lambda from '@aws-cdk/aws-lambda';
-import { FileSystem } from '@aws-cdk/core';
+import { Annotations, FileSystem } from '@aws-cdk/core';
+import { debugModeEnabled } from '@aws-cdk/core/lib/debug';
+import * as cxapi from '@aws-cdk/cx-api';
 import { Construct } from 'constructs';
 
 /**
  * An AWS Lambda layer that includes the AWS CLI.
  */
 export class AwsCliLayer extends lambda.LayerVersion {
+  /**
+   * @internal
+   */
+  public static _tryLoadPackage(targetVersion: string, logs: string[]): any {
+    let availableVersion;
+    let assetPackagePath;
+    try {
+      assetPackagePath = require.resolve(`${AwsCliLayer.assetPackageName}`);
+    } catch (err) {
+      logs.push(`require.resolve('${AwsCliLayer.assetPackageName}') failed`);
+      return;
+    }
+    availableVersion = AwsCliLayer.requireWrapper(path.join(assetPackagePath, '../../package.json'), logs).version;
+
+    if (targetVersion === availableVersion) {
```

Comment:
Do we care exactly? Do we need `semver` in here to check range compatibility?

Reply:
Needs to log if the version is there but we rejected it btw. That's good info to know.